packages/rocketchat-lib/server/methods/leaveRoom.js

```diff
@@ -27,7 +26,17 @@ Meteor.methods({
 				throw new Meteor.Error('error-you-are-last-owner', 'You are the last owner. Please set new owner before leaving the room.', { method: 'leaveRoom' });
 			}
 		}
-
+		if (room.automatic) {
+
+			RocketChat.leave_automatic_channel(room.name);
+			// delete room if its the last user
+			if (room.usernames.length===1) {
```

Comment:
There might be some race condition here. What if some other user joined here in meantime?